backend/hct_mis_api/apps/periodic_data_update/service/PeriodicDataUpdateTemplateService.py

```diff
@@ -0,0 +1,98 @@
+    TicketSystemFlaggingDetails,
+    TicketIndividualDataUpdateDetails,
+    TicketSensitiveDetails,
+    TicketComplaintDetails,
+    TicketDeleteIndividualDetails,
+)
+from hct_mis_api.apps.household.models import Individual
+from hct_mis_api.apps.payment.models import Payment
+from hct_mis_api.apps.periodic_data_update.models import PeriodicDataUpdateTemplate
+
+
+class PeriodicDataUpdateExportTemplateService:
+    def __init__(self, periodic_data_update_template: PeriodicDataUpdateTemplate):
+        self.periodic_data_update_template = periodic_data_update_template
+        self.program = periodic_data_update_template.program
+        self.registration_data_import_id_filter = periodic_data_update_template.filters.get(
+            "registration_data_import_id"
+        )
+        self.target_population_id_filter = periodic_data_update_template.filters.get("target_population_id")
+        self.gender_filter = periodic_data_update_template.filters.get("gender")
+        self.age_filter = periodic_data_update_template.filters.get("age")
+        self.registration_date_filter = periodic_data_update_template.filters.get("registration_date")
+        self.has_grievance_ticket_filter = periodic_data_update_template.filters.get("has_grievance_ticket")
+        self.admin1_filter = periodic_data_update_template.filters.get("admin1")
+        self.admin2_filter = periodic_data_update_template.filters.get("admin2")
+        self.received_assistance_filter = periodic_data_update_template.filters.get("received_assistance")
+
+    def _get_individuals_queryset(self):
+        queryset = Individual.objects
+        queryset = queryset.filter(program=self.program)
+        if self.registration_data_import_id_filter:
+            queryset = queryset.filter(registration_data_import_id=self.registration_data_import_id_filter)
+        if self.target_population_id_filter:
+            queryset = queryset.filter(target_population_id=self.target_population_id_filter)
+        if self.gender_filter:
+            queryset = queryset.filter(sex=self.gender_filter)
+        if self.age_filter:
+            age_from = self.age_filter.get("from")
+            age_to = self.age_filter.get("to")
+            q = age_to_birth_date_query("RANGE", [age_from, age_to])
+            queryset = queryset.filter(q)
+        if self.registration_date_filter:
+            registration_date_from = self.registration_date_filter.get("from")
+            registration_date_to = self.registration_date_filter.get("to")
+            queryset = queryset.filter(first_registration_date__range=[registration_date_from, registration_date_to])
+
+        if self.admin1_filter:
+            queryset = queryset.filter(admin1__in=self.admin1_filter)
+        if self.admin2_filter:
+            queryset = queryset.filter(admin2__in=self.admin2_filter)
+        if self.has_grievance_ticket_filter:
+            queryset = self._get_grievance_ticket_filter(queryset)
```

Comment:
remember this value can also be `False` - individuals not in the `ticket_individual_ids`